hnn_core/gui/gui.py

```diff
@@ -1963,7 +2031,7 @@ def update_common_dendrite_sections(sections, mechs_params):
         name for name in sections.keys() if name != 'soma'
     ]
     for section in dendrite_sections:
-        sections[section].mechs.update(mechs_params)
+        sections[section].mechs.update(deepcopy(mechs_params))
```

Comment:
Is this deepcopy necessary ? This operations hit the performance of the app

Reply:
it is necessary because a nested dictionary is being added, I tried a regular `.copy()` and found that the inner dictionary was still pointing to the same variable

This is only necessary for the `ar` mechanism since `gbar_ar` is a partial function who's outputs depend on the geometry of each `section`

Reply:
I'll revisit this in a future PR. I can improve this piece of code for sure.